### DISCUSSION TITLE
ElastoDyn

### DISCUSSION THREAD

**Discussion by Unique-Wang** [OpenFAST/openfast#1065] [Q&A] [answered]
Dear @jjonkman,

As you can see,there is a big **YawBrMy** discrepancy between OpenFAST model and Bladed model ,while **YawBrMx**,**YawBrMz**,**YawBrFx**,**YawBrFy** and **YawBrFz** basically match. I presume that it is because the coordinate systems are different. Do you think so? And if so ,what can I do with the elastodyn module to shrink the discrepancy?
![My差异图](https://user-images.githubusercontent.com/87693817/160342122-ff66d32b-3fef-4843-bc77-eac1d83dbf54.png)
![企业微信截图_20220328142003](https://user-images.githubusercontent.com/87693817/160342134-4d38013f-6f2e-4064-80fe-251c4f9d0cdd.png)
Thank you!

Best regards,

**Comment by jjonkman** [comment, marked as answer]
Dear @Unique-Wang,

Is the line labeled "Yaw bearing My New" the output from Bladed?

The y bending moment at the yaw bearing is typically driven by a combination of the moment arm between the yaw bearing and hub crossed with the rotor thrust and asymmetric loading of the rotor (e.g., from shear).  Are these set consistently between the OpenFAST and Bladed models?

Best regards,

**Comment by Unique-Wang** [comment]
Dear @jjonkman ，

Thank you for your answer. I'm thinking maybe it's because I didn't set **overhang** to a reasonable value that caused **Yaw bearing My** so different. 
And there's another question, I wanna use BModes to calculate modeshapes for my tower, but in the **bmi** file, I didn't know how to get the **ixx_tip**, **iyy_tip** and **izz_tip** in the **bmi** file, I found this https://forums.nrel.gov/t/bmodes-about-the-tower-top-mass-properties/1971,but its for FASTV7 instead of OpenFAST,  I tried to do the linearization with OpenFAST hoping to get the 6X6 mass matrix but failed, so confused. What's your suggestion?

Thank you!

Best regards,

**Reply by jjonkman** [reply]
Dear @Unique-Wang,

It is true that the linearization functionality of OpenFAST cannot directly output the rigid-body mass matrix that was possible with FAST v7.  However, the rigid-body mass matrix can often be inferred from other matrices generated through the linearization process.  This has been discussed in the following forum topic: https://forums.nrel.gov/t/openfast-2nd-order-linearization/2249.

Best regards,

**Reply by ebranlard** [reply]
In my repository of scripts (welib), I have automated the extraction of full wind turbine inertia or RNA inertia using OpenFAST linearizations. The RNA inertia is what you need as input to BModes. You can find an example here: 
      https://github.com/ebranlard/welib/blob/dev/welib/fast/examples/Example_ExtractInertia.py
You can look at the files generated in the working directory to see how the input files are changed, or look into the source code:  
     https://github.com/ebranlard/welib/blob/dev/welib/fast/extract.py
The example script also contains a python-only method (YAMS). It returns values that are close but I would consider this a work in progress.  To use these scripts you'll need to install the dev version of welib. If you encounter issues with this script, please post them in the welib repository to avoid interfering with this discussion.

**Reply by Unique-Wang** [reply]
Dear @ebranlard，

Solved! Thank you so much!

Best regards,

**Reply by Unique-Wang** [reply]
Dear @jjonkman ，

Thank you for your answer! 
And I remember you posted an article you wrote about the calculation principle of rotor inertia( or about Mx, My and Mz? Don't remeber anymore...........) in a forum topic but I forgot which one, been spending the whole afternoon to find it but failed, could you please send it to me here or tell me how to find those articles you wrote on NREL?
Thank you in advance!

Best regards,

**Reply by jjonkman** [reply]
Dear @Unique-Wang,

I'm sorry, but I'm not sure which forum topic you are referring to.

Best regards,

**Reply by Unique-Wang** [reply]
Dear @jjonkman ,

Could you please send me the seismic module with source code you are talking about in this post? https://forums.nrel.gov/t/running-examples-with-the-seismic-module/1846/54. My email : wangweiunique@sina.com
Thank you!

Best regards,

**Reply by jjonkman** [reply]
Dear @Unique-Wang,

I just sent you the Seismic archive of FAST v7 via WeTransfer.

Best regards,

**Reply by Unique-Wang** [reply]
Got it ,thank you!
